### PR TITLE
Disable Attachment Routing

### LINKED ISSUE #10: Suppress Media Item Detail Pages by Default
It is rarely the case that we want an attachment to have its own detail page publicly accessible, so this functionality should be turned off by default.

### PR DESCRIPTION
**Pull request by srtfisher**
- Remove the attachment page from the front-end, including from the admin modal and the admin bar.
- Plugs in a default value to the `wp_attachment_pages_enabled` option introduced in 6.4.

Resolves #10

### CODE REVIEW

File: src/alley/wp/alleyvate/features/class-disable-attachment-routing.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+			status_header( 404 );
+		}
+	}
+
+	/**
+	 * Remove attachment link from admin bar.
+	 *
+	 * @param \WP_Admin_Bar $wp_admin_bar Admin bar class.
+	 */
+	public static function action__admin_bar_menu( \WP_Admin_Bar $wp_admin_bar ) {
```
Comment (renatonascalves) [inline src/alley/wp/alleyvate/features/class-disable-attachment-routing.php:82]:
```suggestion
	 * @param \WP_Admin_Bar $wp_admin_bar Admin bar class.
	 */
	public static function action__admin_bar_menu( $wp_admin_bar ): void {
```

File: tests/alley/wp/alleyvate/features/test-disable-attachment-routing.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		parent::setUp();
+
+		$this->feature = new Disable_Attachment_Routing();
+	}
+
+	/**
+	 * Test that the attachment permalink is empty.
+	 */
+	public function test_attachment_permalink(): void {
+		$attachment_id = $this->factory()->attachment->create();
```
Comment (renatonascalves) [inline tests/alley/wp/alleyvate/features/test-disable-attachment-routing.php:45]:
```suggestion
		$attachment_id = self::factory()->attachment->create();
```

File: tests/alley/wp/alleyvate/features/test-disable-attachment-routing.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		$this->feature->boot();
+
+		$this->assertEmpty( get_permalink( $attachment_id ) );
+	}
+
+	/**
+	 * Test that the attachment page returns a 404.
+	 */
+	public function test_attachment_page(): void {
+		$attachment_id = $this->factory()->attachment->create();
```
Comment (renatonascalves) [inline tests/alley/wp/alleyvate/features/test-disable-attachment-routing.php:58]:
```suggestion
		$attachment_id = self::factory()->attachment->create();
```

**Review by renatonascalves — APPROVED** [review]
🌵

**Review by kevinfodness — APPROVED** [review]
🍣

**Pull request merged**